lib/sidebar.rb

```diff
@@ -29,185 +29,193 @@ def pre_render(page)
           header = {
             id: header['id'],
             title: header.content.strip,
-            hash: (child['href']).to_s
+            hash: (child_markup['href']).to_s
           }
           headers.push(header)
         end
-        sanitized_filename = sanitize_filename(filename)
-        @filename_with_headers[sanitized_filename] = { headers: headers }
-      end
 
-      Dir.glob("#{site.config['destination']}/**/*.html") do |filename|
-        sanitized_filename = sanitize_filename(filename)
-        sidebar = render(sanitized_filename)
-        file = File.open(filename) { |f| Nokogiri::HTML(f) }
-        file.xpath('//*[@id="dx-sidebar-main-nav-ul"]').each do |location|
-          location.inner_html = sidebar
-        end
-        File.open(filename, 'w') { |f| f.write(file.to_html(encoding: 'UTF-8')) }
+        @pages_with_headers[sanitized_filename] = { headers: headers }
       end
 
-      # File.open('_site/sidebar.html', 'w') { |f| f.write(sidebar) }
+      files
     end
 
-    private
+    def write_sidebar_to_files(files)
+      files.each do |file|
+        sanitized_filename = file[:sanitized_filename]
+        filename = file[:filename]
+        doc = file[:doc]
+        sidebar_markup = nil
+
+        begin
+          sidebar_markup = render(sanitized_filename)
+        rescue StandardError => e
+          Jekyll.logger.error("           Sidebar: Unable to render sidebar_markup for '#{filename}'.")
+          Jekyll.logger.debug("           Sidebar: #{e.message}. #{e.backtrace.inspect}")
+          next
+        end
+
+        sidebar_containers = doc.xpath('//*[@id="dx-sidebar-main-nav-ul"]')
+
+        unless sidebar_containers.any?
+          Jekyll.logger.debug("           Sidebar: No sidebar container found in #{filename}")
+          next
+        end
+
+        sidebar_containers.each do |sidebar_container|
+          sidebar_container.inner_html = sidebar_markup
+        end
 
-    def sanitize_filename(filename)
-      sanitized_filename = filename.match(/(?m)(?<=\b_site).*$/)[0]
-      sanitized_filename = sanitized_filename.gsub('index.html', '')
-      sanitized_filename.gsub('.html', '')
+        Jekyll.logger.debug("   Writing Sidebar: #{filename}")
+        File.open(filename, 'w') { |f| f.write(doc.to_html(encoding: 'UTF-8')) }
+      end
     end
 
-    def generateSubgroup(filename, key, value, all_subgroups, level)
-      title = value[:title].split('–').last
-      
-      subsubgroup_list = all_subgroups.select do |subsubgroup_key, _subsubgroup_value|
-        subsubgroup_key.include? key and subsubgroup_key != key and \
-          key.split('/').length > level
+    def generate_sub_group(filename, path, page, all_child_pages, level)
+      title = page[:title].split('–').last.strip
+
+      url = page[:url]
+      headers = page[:headers]
+
+      grandchildren = all_child_pages.select do |grandchild_path, _|
+        grandchild_path.include? path and grandchild_path != path and \
```

Comment:
Codacy found an issue: [Use `&&` instead of `and`.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=6198407)